SetupDataPkg/Library/ConfigVariableListLib/ConfigVariableListLib.c

```diff
@@ -96,98 +300,55 @@ ParseActiveConfigVarList (
     goto Exit;
   }
 
-  while (ListIndex < BinSize) {
+  while (ListIndex < VariableListBufferSize) {
     // index into variable list
-    VarList = (CONFIG_VAR_LIST_HDR *)((CHAR8 *)BlobPtr + ListIndex);
+    VarList = (CONST CONFIG_VAR_LIST_HDR *)((CHAR8 *)VariableListBuffer + ListIndex);
+
+    LeftSize = VariableListBufferSize - ListIndex;
+    Status   = ConvertVariableListToVariableEntry (VarList, &LeftSize, &(*ConfigVarListPtr)[*ConfigVarListCount]);
 
-    if (ListIndex + sizeof (*VarList) + VarList->NameSize + VarList->DataSize + sizeof (*Guid) +
-        sizeof (Attributes) + sizeof (CRC32) > BinSize)
-    {
+    if (EFI_ERROR (Status)) {
       // the NameSize and DataInBinSize have bad values and are pushing us past the end of the binary
-      DEBUG ((DEBUG_ERROR, "%a Configuration VarList had bad NameSize or DataInBinSize\n", __FUNCTION__));
+      DEBUG ((DEBUG_ERROR, "%a Configuration VarList conversion failed %r\n", __FUNCTION__, Status));
       ASSERT (FALSE);
-      Status = EFI_COMPROMISED_DATA;
       break;
     }
 
-    /*
-     * Var List is in DmpStore format:
-     *
-     *  struct {
-     *    CONFIG_VAR_LIST_HDR VarList;
-     *    CHAR16 Name[VarList->NameSize/2];
-     *    EFI_GUID Guid;
-     *    UINT32 Attributes;
-     *    CHAR8 DataInBin[VarList-DataSize];
-     *    UINT32 CRC32; // CRC32 of all bytes from VarList to end of DataInBin
-     *  }
-     */
-    NameInBin  = (CHAR16 *)(VarList + 1);
-    Guid       = (EFI_GUID *)((CHAR8 *)NameInBin + VarList->NameSize);
-    Attributes = *(UINT32 *)(Guid + 1);
-    DataInBin  = (CHAR8 *)Guid + sizeof (*Guid) + sizeof (Attributes);
-    CRC32      = *(UINT32 *)(DataInBin + VarList->DataSize);
-
-    // on next iteration, skip past this variable
-    ListIndex += sizeof (*VarList) + VarList->NameSize + VarList->DataSize + sizeof (*Guid) + sizeof (Attributes)
-                 + sizeof (CRC32);
-
-    VarName = AllocatePool (VarList->NameSize);
-    if (VarName == NULL) {
-      DEBUG ((DEBUG_ERROR, "%a Failed to allocate memory for VarName size: %u\n", __FUNCTION__, VarList->NameSize));
-      Status = EFI_OUT_OF_RESOURCES;
-      goto Exit;
-    }
-
-    CopyMem (VarName, NameInBin, VarList->NameSize);
+    ListIndex += LeftSize;
 
     // This check needs to come after we allocate VarName so we can compare against a 16 bit aligned
     // string and not assert

```

Comment:
comment out of date

Reply:
Lines got moved around and this comment does not fully reflect the original change. I think I updated the correct line of comment for this feedback.